controllers/etcd_controller.go

```diff
@@ -1045,3 +1055,24 @@ func (r *EtcdReconciler) claimConfigMaps(etcd *druidv1alpha1.Etcd, selector labe
 	cm := NewEtcdDruidRefManager(r, etcd, selector, etcdGVK, canAdoptFunc)
 	return cm.ClaimConfigMaps(ss)
 }
+
+// SetupWithManager sets up manager with a new controller and r as the reconcile.Reconciler
+func (r *EtcdReconciler) SetupWithManager(mgr ctrl.Manager, workers int, ignoreOperationAnnotation bool) error {
+	predicates := []predicate.Predicate{
+		druidpredicates.GenerationChangedPredicate{},
+		druidpredicates.LastOperationNotSuccessful(),
+	}
+	builder := ctrl.NewControllerManagedBy(mgr).WithOptions(controller.Options{
+		MaxConcurrentReconciles: workers,
+	})
+	if !ignoreOperationAnnotation {
+		predicates = append(predicates, druidpredicates.HasOperationAnnotation())
+	}
+	builder = builder.WithEventFilter(druidpredicates.Or(predicates...))
```

Comment:
Can you check if the deletion case works properly?
Because here is what I think can happen:
- The object is deleted 
- `deletionTimestamp` is set, no `generation` change
- Update event is triggered (due to `finalizer` the deletion event is not triggered immediately)
- All predicates return `false`

Also check the following scenario:
- Delete the resource
- Controller starts deletion operation
- Interrupt/restart controller while deletion operation is running
- Make sure controller continues after restart (now it gets a `create` event for an object with a `deletionTimestamp`.)

Reply:
@timuthy see https://github.com/gardener/etcd-druid/pull/33#discussion_r399212827 - we figured out that the generation is bumped when the deletion timestamp gets set. Hence, the first scenario should be covered.

The second scenario should also be covered because the `observedGeneration` is only updated at the end of a successful reconciliation, or after an error occurred (but then the `lastError` will be set as well, allowing to pick the `Etcd` up again). WDYT?

Reply:
Okay thanks @rfranzke sounds good to me.